src/middlewared/middlewared/plugins/system/vendor/vendor.py

```diff
@@ -0,0 +1,28 @@
+import json
+import os
+
+from middlewared.api import api_method
+from middlewared.api.current import VendorNameArgs, VendorNameResult, UnvendorArgs, UnvendorResult
+from middlewared.service import Service
+
+
+SENTINEL_FILE_PATH = '/data/.vendor'
+
+
+class VendorService(Service):
+
+    @api_method(VendorNameArgs, VendorNameResult, private=True)
+    def name(self) -> str | None:
+        try:
+            with open(SENTINEL_FILE_PATH, 'r') as file:
+                if contents := file.read():
+                    return json.loads(contents).get('name')
+        except FileNotFoundError:
```

Comment:
You beat me to it but let's catch the `JSONDecodeError` as well.

Reply:
You can do `except FileNotFoundError json.JSONDecodeError)`

Reply:
Wasn't sure if it would be desirable to.